src/ray/raylet/worker_pool.cc

```diff
@@ -209,8 +203,8 @@ std::shared_ptr<Worker> WorkerPool::PopWorker(const TaskSpecification &task_spec
   std::shared_ptr<Worker> worker = nullptr;
   if (actor_id.is_nil()) {
     if (!state.idle.empty()) {
-      worker = std::move(state.idle.back());
-      state.idle.pop_back();
+      worker = std::move(*state.idle.begin());
+      state.idle.erase(state.idle.begin());
```

Comment:
@atumanov does this look right?

Reply:
LGTM

Reply:
@ericl , yeah this is fine. std::unordered_set::extract (C++17) would've been ideal here.
Confirmed with : 
```
valgrind --tool=memcheck --leak-check=full --show-leak-kinds=all --track-origins=yes ./container
```
```
#include <iostream>
#include <unordered_set>

class MyClass {
 public:
  MyClass() {}
  MyClass(int i): myvalue_(i) {}

  int GetValue() const {
    return myvalue_;
  }

  void SetValue(int newvalue) {
    myvalue_ = newvalue;
  }

  bool operator==(const MyClass &other) {
    return myvalue_ == other.myvalue_;
  }

  friend bool operator==(const MyClass &obj1, const MyClass &obj2) {
    return obj1.GetValue() == obj2.GetValue();
  }

  ~MyClass() {}
 private:
  int myvalue_ = 0;
};

namespace std {
template<>
struct hash<MyClass> {
size_t operator()(MyClass const &obj) const {
  return obj.GetValue();
}
};
}

using namespace std;
int main() {
  unordered_set<MyClass> myintobjset;

  for (int i=0; i<100; i++) {
    myintobjset.emplace(MyClass(i));
  }

  MyClass myintobj = std::move(*myintobjset.begin());
  myintobjset.erase(myintobjset.begin());
  cout << "my saved int obj " << myintobj.GetValue() << endl;

  for (const auto &elem : myintobjset) {
    cout << elem.GetValue() << " ";
  }
  cout << endl;

  return 0;
}
```

Reply:
Thanks for trying :)